reactor-core/src/main/java/reactor/core/publisher/FluxWindowPredicate.java

```diff
@@ -348,6 +348,7 @@ else if (!outputFused) {
 						}
 
 						if (WIP.decrementAndGet(this) == 0) {
+							s.cancel();
```

Comment:
this is susceptible to racing with `onComplete`/`onError`. It's not a big big issue, but in order to reduce the surface area for that race you could add an `if (!done)` condition on the `s.cancel()`.